src/common/tasks_persistence.py

```diff
@@ -60,7 +60,10 @@ def save_task(r,id,task):
     try:
         # serialize json from task:
         jtask = json.dumps(task)
-        r.set(task_id,jtask)
+        if exp_time:
+            r.setex(task_id,exp_time,jtask)
+        else:
+            r.set(task_id,jtask)
```

Comment:
I think we can use r.set(task_id,jtask, ex=exp_time) . According to the documentation at https://redis-py.readthedocs.io/en/stable/